### PR TITLE
Fixes #227

### LINKED ISSUE #227: Add logging to webhook.js
Currently, webhook.js does not log in the same way as the rest of the application (using bunyan).  It should use `req.log` to do this instead of `console.log`

### PR DESCRIPTION
**Pull request by unknowndomain**

### CODE REVIEW

**Comment by unknowndomain** [comment]
@naxxfish No one else is really able to do the code review necessary.

File: webhook.js [hunk 1]
```diff
@@ -323,28 +402,61 @@ function handleMandateResourceEvent( event ) {
 			// Do nothing, we already have the details on file.
 			break;
 		case 'reinstated':
-			console.log( 'Mandate reinstated, its likely this mandate wont be linked to a member...' )
-			console.log( event );
+			log.info( {
+				app: 'webhook',
+				action: 'mandate-reinstated',
```
Comment (naxxfish) [inline webhook.js]: `reinstate-mandate`

File: webhook.js [hunk 2]
```diff
@@ -323,28 +402,61 @@ function handleMandateResourceEvent( event ) {
...
 }
 
 function cancelledMandate( event ) {
 	Members.findOne( { 'gocardless.mandate_id': event.links.mandate }, function( err, member ) {
-		if ( ! member ) return console.log( 'Mandate ID: ' + event.links.mandate + ' was not associated with a user' );
-		console.log( 'Removing mandate ID: ' + event.links.mandate + ' for user: ' + member.email );
+		if ( ! member ) {
+			log.info( {
+				app: 'webhook',
+				action: 'unlinked-mandate',
```
Comment (naxxfish) [inline webhook.js]: `unlink-mandate`

File: webhook.js [hunk 3]
```diff
@@ -182,14 +206,25 @@ function extendMembership( event ) {
 						member.permissions[p].date_expires.setMinutes( 0 );
 						member.permissions[p].date_expires.setSeconds( 0 );
 						member.permissions[p].date_expires.setMilliseconds( 0 );
-						console.log( 'Extending "' + member.email + '" membership permission until: ' + member.permissions[p].date_expires );
+						log.info( {
+							app: 'webhook',
+							action: 'membership-extended',
```
Comment (naxxfish) [inline webhook.js]: `extend-membership` ? (verb should come first)

File: webhook.js [hunk 4]
```diff
@@ -143,25 +139,53 @@ function createPayment( event ) {
...
 	Payments.findOne( { payment_id: event.links.payment }, function( err, payment ) {
 		if ( ! payment ) return; // There's nothing left to do here.
 		payment.status = event.details.cause;
 		payment.updated = new Date();
 		payment.save( function( err ) {
-			if ( err ) console.log( err );
+			if ( err ) {
+				log.debug( {
+					app: 'webhook',
+					action: 'payment-update-error',
```
Comment (naxxfish) [inline webhook.js]: `error-updating-payment`? (the main verb is error)

File: webhook.js [hunk 5]
```diff
@@ -143,25 +139,53 @@ function createPayment( event ) {
...
+					error: err
+				} );
+			} else {
+				log.info( {
+					app: 'webhook',
+					action: 'payment-update',
```
Comment (naxxfish) [inline webhook.js]: `update-payment` ?

File: webhook.js [hunk 6]
```diff
@@ -143,25 +139,53 @@ function createPayment( event ) {
...
+					if ( err ) {
+						log.debug( {
+							app: 'webhook',
+							action: 'payment-creation-error',
+							error: err
+						} );
+					} else {
+						log.info( {
+							app: 'webhook',
+							action: 'payment-created',
```
Comment (naxxfish) [inline webhook.js]: `create-payment`?

File: webhook.js [hunk 7]
```diff
@@ -143,25 +139,53 @@ function createPayment( event ) {
 					payment.member = member._id;
 				}
 				new Payments( payment ).save( function( err ) {
-					if ( err ) console.log( err );
+					if ( err ) {
+						log.debug( {
+							app: 'webhook',
+							action: 'payment-creation-error',
```
Comment (naxxfish) [inline webhook.js]: `error-creating-payment` ?

**Pull request merged**